### ISSUE TITLE
TD-SCDMA not available (or broken?)

### ISSUE THREAD

**Issue by emfox** [zwliew/android_device_xiaomi_hm2014811#51]
2014813, my sim card of China Mobile is for 2G/3G only (no 4G,because the card is old and not upgraded), but after flash this rom, I can only get EGDE signal, on which area I can reach TD-SCDMA on stock MIUI6 before. I've tried both LTE and 3G on Preferred network type of Mobile network settings, but neither works.

**Comment by zwliew** [comment]
Working fine for me.

**Comment by emfox** [comment]
I borrow a China Mobile 4G sim card and try to test it. by switching Preferred network type in LTE/3G/2G, I got the signal letters of 4G/E/E, so still no H(3G) at all.

**Comment by zwliew** [comment]
Anyone else has this issue on TD-SCDMA?

**Comment by zwliew** [comment]
@enfox Try changing the ro.telephony.default_network property in build.prop from 9 to 20.

Change:
ro.telephony.default_network=9,1
to:
ro.telephony.default_network=20,1

**Comment by emfox** [comment]
tried, but not work. 

Any other step need to be done beside rebooting?

**Comment by ankkitnegi** [comment]
Can anybody give me the link of zwliew cm12

**Comment by emfox** [comment]
You're just here at your desired page.

**Comment by zwliew** [comment]
Does this still persist in CM 12.1?

**Comment by emfox** [comment]
I haven't tried your cm12 build since the camera is not ready, and I need camera for daily use.

**Comment by zwliew** [comment]
The camera will be fixed in a build released today, so do try it out when you have the time.

**Comment by emfox** [comment]
I've test the new build of cm12 0925 and yes this bug is still there. 
To be in detail, I change default_network from 9,1 to 20,1 and reboot, the network type is still LTE/GSM auto, if I switch system setting from LTE to 3G,  network type is changed to WCDMA preferred. I have to manually enter phone test mode(*#*#4636#*#*), choose TD-SCDMA/anything, then it works.

**Comment by zwliew** [comment]
I believe I know how to fix the problem, but in doing so I may break LTE for GSM devices. I'm not sure how to make changes for different models of devices.

**Comment by adrianardizza** [comment]
you should make 2 separate builds maybe, 1 for 2014811/2014817 and 1 for 2014813 which uses TDD-SCDMA and TDD-LTE.

**Comment by emfox** [comment]
No, that's not true, although 2014813 is for China Mobile while for sale, and they said it use TD-SCDMA and TDD-LTE, but when you get the phone, simply enable a networking switch in MIUI setting menu, the phone will soon become available for FDD-LTE and WCDMA. that is, a FDD/TDD/WCDMA/TD-SCDMA/GSM enabled phone.
